packages/js/components/src/form/form.tsx

```diff
@@ -96,6 +114,23 @@ function Form< Values extends Record< string, any > >(
 		// eslint-disable-next-line @typescript-eslint/no-explicit-any
 		( name: string, value: any ) => {
 			setValues( { ...values, [ name ]: value } );
+			if (
+				props.initialValues[ name ] !== value &&
+				! changedFields[ name ]
+			) {
+				setChangedFields( {
+					...changedFields,
+					[ name ]: true,
+				} );
+			} else if (
+				props.initialValues[ name ] === value &&
+				changedFields[ name ]
+			) {
+				setChangedFields( {
+					...changedFields,
+					[ name ]: false,
+				} );
+			}
```

Comment:
Added `changedFields` logic so we can determine if the form is `dirty` (has been changed) or not.